### ISSUE TITLE
SimpleCache returns stale data from a recycled resource instance URI

### ISSUE THREAD

**Issue by swiharta** [django-tastypie/django-tastypie#438]
TL;DR: If you POST, DELETE, re-POST (re-instantiating the same URI) within SimpleCache's 60s timeout, a GET request to that URI sends back stale data from the first POST.

I'm making a calendar app with Backbone and I just got done debugging an issue with a ModelResource that I had enabled `SimpleCache` on. I should have taken it off anyway, as it seems to impart no significant performance gain in particular when fetching a large amount of data, where you could actually use the speed

My issue is that when I did a POST to a ModelResource (followed automatically via Backbone.sync with a GET request to the new URI to pull down the new `id`), then immediately did a DELETE to the new instance's URI, and then quickly another POST / GET cycle (re-using the same URI), Tastypie was apparently sending back stale, cached data from the first POST. This had the effect of rendering calendar event data from my first POST, while saving the correct second POST data to the database.

This was all done through AJAX with single model `add` / `destroy` commands from my Backbone app. When I very quickly did a full page refresh, a Backbone `fetch` to my ModelResource's root url pulled down the correct, latest data that I had POST'ed for that resource.

I finally realized some form of caching was the root of the issue, because if I waited long enough between creating and deleting / recreating a resource (and without doing a full page refresh), there was no issue. 

Turning off SimpleCache on my ModelResource completely eliminated this problem (turning it back on just now, the bug returns). It was not, as I had considered, an issue with browser caching. 

As I said, when hitting the root url of my ModelResource, the correct, new data was available immediately. Only the recycled URI of a resource instance was sending back stale data.

I'd also love to be enlightened about a case where SimpleCache gives a tangible performance benefit.

EDIT: Now I take a second to actually look at the code for SimpleCache and of course it has a 60s timeout. Granted the cache is only stale when I delete a URI and recycle it within the cache timeout. I know this is just a "simple" cache, but this still seems like buggy enough behavior to report.
